packages/cli/configuration/src/generators-yml/getGeneratorVersions.ts

```diff
@@ -1,38 +1,140 @@
-    // eslint-disable-next-line @typescript-eslint/dot-notation
-    const generatorVersion = image.Config.Labels?.["version"];
-    if (generatorVersion == null) {
-        context?.logger.error(`No version found behind generator ${generatorName} at tag latest.`);
-        return;
+        default: {
+            context?.logger.warn(`Unrecognized generator name found, attempting to parse manually: ${generatorName}`);
+            if (generatorName.startsWith("fern-")) {
+                return generatorName.replace("fern-", "");
+            }
+            return generatorName;
+        }
     }
-    context?.logger.debug(`Found image behind generator ${generatorName} at tag latest: ${generatorVersion}.`);
-    return generatorVersion;
+    // switch (generatorName) {
+    //     // Python
+    //     case "fern-python-sdk":
+    //         return { language: "python", generatorType: { type: "sdk" } };
+    //     case "fern-pydantic-model":
+    //         return { language: "python", generatorType: { type: "model" } };
+    //     case "fern-fastapi-server":
+    //         return { language: "python", generatorType: { type: "server" } };
+    //     // TypeScript
+    //     case "fern-typescript-browser-sdk":
+    //     case "fern-typescript-node-sdk":
+    //     case "fern-typescript-sdk":
+    //         return { language: "typescript", generatorType: { type: "sdk" } };
+    //     case "fern-typescript-express":
+    //         return { language: "typescript", generatorType: { type: "server" } };
+    //     // Java
+    //     case "fern-java-sdk":
+    //         return { language: "java", generatorType: { type: "sdk" } };
+    //     case "java-model":
+    //         return { language: "java", generatorType: { type: "model" } };
+    //     case "fern-java-spring":
+    //         return { language: "java", generatorType: { type: "server" } };
+    //     // Go
+    //     case "fern-go-sdk":
+    //         return { language: "go", generatorType: { type: "sdk" } };
+    //     case "fern-go-model":
+    //         return { language: "go", generatorType: { type: "model" } };
+    //     case "fern-go-fiber":
+    //         return { language: "go", generatorType: { type: "server" } };
+    //     // C#
+    //     case "fern-csharp-sdk":
+    //         return { language: "csharp", generatorType: { type: "sdk" } };
+    //     case "fern-csharp-model":
+    //         return { language: "csharp", generatorType: { type: "model" } };
+    //     // Ruby
+    //     case "fern-ruby-sdk":
+    //         return { language: "ruby", generatorType: { type: "sdk" } };
+    //     case "fern-ruby-model":
+    //         return { language: "ruby", generatorType: { type: "model" } };
+    //     // Misc.
+    //     case "fern-postman":
+    //         return { language: "postman", generatorType: "other" };
+    //     case "fern-openapi":
+    //         return { language: "openapi", generatorType: "other" };
+    // }
```

Comment:
should we uncomment and leave in a separate file? that way if someone adds php they'll be forced to update this function as well and we can swap during cli v2

Reply:
I removed this since it was legacy! But the above switch should achieve the same thing
The `generatorName` is just a string in Fern today, ideally it'd be an enum or string options to be able to fail when someone adds more languages, but that'd require a few changes throughout the generator yml setup